src/lib/smartlist_core/smartlist_foreach.h

```diff
@@ -83,6 +83,14 @@
          ++var ## _sl_idx) {                                    \
       var = (sl)->list[var ## _sl_idx];
 
+#define SMARTLIST_FOREACH_REVERSE_BEGIN(sl, type, var)  \
```

Comment:
Let's have a unit test for this.

Also, let's document that it isn't compatible with SMARTLIST_DEL_CURRENT

Reply:
Done in 4c44d7b116 and 072e7c87e8 .

Reply:
Done in 4c44d7b116 and 1b706523ee.